plugins/modules/win_gpo_link.ps1

```diff
@@ -0,0 +1,278 @@
+                $mygpo = get-gpo -guid $gpoguid -domain $domain -server "$($checkdc)$($domain)$null"
+                if ($null -ne $mygpo ) {
+                    $gponame = $MyGPO.displayname
+                    $gpodomain = $domain
+                } else {
+                    $gponame = "Orphaned GPLink"
+                    $gpodomain = $domain
+                }
+                #we test the last 2 charaters of the split do determine the status of the GPO link
+                if (($split[$s].endswith(";0"))) {
+                    $enforced = "No"
+                    $enabled = "Yes"
+                } elseif (($split[$s].endswith(";1"))) {
+                    $enabled = "No"
+                    $enforced = "No"
+                } elseif (($split[$s].endswith(";2"))) {
+                    $enabled = "Yes"
+                    $enforced = "Yes"
+                } elseif (($split[$s].endswith(";3"))) {
+                    $enabled = "No"
+                    $enforced = "Yes"
+                }
+                #we create an object representing each GPOs, its links status and link order
+                $return = New-Object psobject
+                $return | Add-Member -membertype NoteProperty -Name "Target" -Value $target
+                $return | Add-Member -membertype NoteProperty -Name "GPOID" -Value $gpoguid
+                $return | Add-Member -membertype NoteProperty -Name "DisplayName" -Value $gponame
+                $return | Add-Member -membertype NoteProperty -Name "Domain" -Value $gpodomain
+                $return | Add-Member -membertype NoteProperty -Name "Enforced" -Value $enforced
+                $return | Add-Member -membertype NoteProperty -Name "Enabled" -Value $enabled
+                $return | Add-Member -membertype NoteProperty -Name "Order" -Value $order
+                $return
+            }
+        }
+    }
+}
+$spec = @{
+    options = @{
+        state       = @{ type = "str"; choices = "query", "present", "absent"; default = "present" }
+        path        = @{ type = "list"; elements = "str"; required = $true }
+        gponame     = @{ type = "str"; required = $true }
+        order       = @{ type = "int" }
+        domain      = @{ type = "str" }
+        enforced    = @{ type = "bool"; default = $false }
+        linkenabled = @{ type = "bool"; default = $true }
+    }
+    supports_check_mode = $true
+}
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$state = $module.Params.state
+$path = $module.Params.path
+$gponame = $module.Params.gponame
+$gpoorder = $module.Params.order
+$gpodomain = $module.Params.domain
+$gposerver = $module.Params.server
+$enforcedbool = $module.Params.enforced
+$gpolinkenabledbool = $module.Params.linkenabled
+$ErrorActionPreference = 'Stop'#Error Action
+try {
+    Import-Module GroupPolicy
+} catch {
+    $module.FailJson("win_grouppolicy requires the GroupPolicy  PS module to be installed")
```

Comment:
By adding the error record as a 2nd argument to `FailJson` we preserve the error details and sent it back to the controller so it can display it better for users. Also the module name was incorrect.

```suggestion
    $module.FailJson("win_gpo_link requires the GroupPolicy  PS module to be installed", $_)
```